### PR TITLE
refactor: Amazon Q Transform service utils

### PR DESCRIPTION
**Pull request by damntrecky**
## Problem

Monolith files are making adding new features difficult without separation of concerns.

## Solution

Refactor our monlithic file into smaller, semantically digestible files 

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*!
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts:1]: Note: We have plans to create a true `FileManager` class. That is outside the scope of this PR.

File: packages/core/src/test/codewhisperer/commands/transformByQ.test.ts [hunk 2]
```diff
@@ -77,7 +78,7 @@ describe('transformByQ', function () {
     })
 
     it('WHEN validateProjectSelection called on non-Java project THEN throws error', async function () {
-        const dummyCandidateProjects: TransformationCandidateProject[] = [
+        const dummyCandidateProjects: model.TransformationCandidateProject[] = [
```
Comment (damntrecky) [inline packages/core/src/test/codewhisperer/commands/transformByQ.test.ts:81]: Nit: Move this to straight import

File: packages/core/src/test/codewhisperer/commands/transformByQ.test.ts [hunk 3]
```diff
@@ -94,7 +95,7 @@ describe('transformByQ', function () {
         await toFile('', dummyPath)
         const findFilesStub = sinon.stub(vscode.workspace, 'findFiles')
         findFilesStub.onFirstCall().resolves([folder.uri])
-        const dummyCandidateProjects: TransformationCandidateProject[] = [
+        const dummyCandidateProjects: model.TransformationCandidateProject[] = [
```
Comment (damntrecky) [inline packages/core/src/test/codewhisperer/commands/transformByQ.test.ts:98]: Nit: Move this to straight import

File: packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { existsSync, writeFileSync } from 'fs'
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts:5]: Question: We should strictly true to use `fs` and not `fs-extra` correct?

**Comment by mk-fan** [comment]
Fix the failed tests.

File: packages/core/src/codewhisperer/service/transformByQ/transformMavenHandler.ts [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+import * as vscode from 'vscode'
+import { FolderInfo, transformByQState } from '../../models/model'
+import { getLogger } from '../../../shared/logger'
+import * as CodeWhispererConstants from '../../models/constants'
+import { spawnSync } from 'child_process' // Consider using ChildProcess once we finalize all spawnSync calls
+import { CodeTransformMavenBuildCommand, telemetry } from '../../../shared/telemetry/telemetry'
+import { codeTransformTelemetryState } from '../../../amazonqGumby/telemetry/codeTransformTelemetryState'
+import { MetadataResult } from '../../../shared/telemetry/telemetryClient'
+import { ToolkitError } from '../../../shared/errors'
+import { throwIfCancelled } from '../securityScanHandler'
```
Comment (mk-fan) [inline packages/core/src/codewhisperer/service/transformByQ/transformMavenHandler.ts:14]: Not sure if this is the right `throwIfCancelled` to use?

File: packages/core/src/test/codewhisperer/commands/transformByQ.test.ts [hunk 6]
```diff
@@ -14,22 +14,24 @@ import * as CodeWhispererConstants from '../../../codewhisperer/models/constants
...
-    TransformationCandidateProject,
-} from '../../../codewhisperer/service/transformByQHandler'
 import path from 'path'
 import { createTestWorkspaceFolder, toFile } from '../../testUtil'
 import {
     NoJavaProjectsFoundError,
     NoMavenJavaProjectsFoundError,
     NoOpenProjectsError,
 } from '../../../amazonqGumby/errors'
+import { throwIfCancelled } from '../../../codewhisperer/service/securityScanHandler'
```
Comment (mk-fan) [inline packages/core/src/test/codewhisperer/commands/transformByQ.test.ts:24]: Not sure if this is the right `throwIfCancelled` to use?

File: packages/core/src/codewhisperer/service/transformByQ/transformMavenHandler.ts (hunk 5 above)
Comment (damntrecky) [reply packages/core/src/codewhisperer/service/transformByQ/transformMavenHandler.ts:14]: BLOCKER: This is the wrong import. Nice catch.

**Review by justinmk3 — APPROVED** [review]
LGTM assuming this is only moving code and not changing logic.

CI has errors that must be fixed.

File: packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts (hunk 4 above)
Comment (justinmk3) [reply packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts:5]: Should use FileSystemCommon: https://github.com/aws/aws-toolkit-vscode/blob/897ef7d03adf117ea5873d3cbc396e394b25c800/packages/core/src/srcShared/fs.ts#L32

File: packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts (hunk 1 above)
Comment (justinmk3) [reply packages/core/src/codewhisperer/service/transformByQ/transformFileHandler.ts:1]: what does it do? there are existing filemanager modules, in the codebase.
